src/nullablevector.jl

```diff
@@ -311,3 +313,27 @@ function Base.empty!(X::NullableVector)
     empty!(X.isnull)
     return X
 end
+
+function vcat(A::AbstractVector, B::AbstractVector...)
```

Comment:
If you implement the two-argument versions, please also implement the three-argument versions. Else the rule is going to be even more complex than what you wrote above.

Of course, this rests on the assumption that concatenating more than 3 arrays is a somewhat rare operation, so that it's not a big issue if other variants behave differently. Might not be the case, but only special-casing two-argument methods is clearly a worse solution.